src/cargo/ops/cargo_clean.rs

```diff
@@ -368,13 +381,35 @@ impl<'cfg> CleanContext<'cfg> {
                     paths::remove_dir_all(entry.path())?;
                 }
             } else {
-                rm_file(entry.path())?;
+                rm_file(entry.path(), entry.metadata())?;
             }
         }
 
         Ok(())
     }
 
+    fn display_summary(&self) -> CargoResult<()> {
+        let status = if self.dry_run { "Summary" } else { "Removed" };
+        let byte_count = if self.total_bytes_removed == 0 {
+            String::new()
+        } else {
+            // Don't show a fractional number of bytes.
+            if self.total_bytes_removed < 1024 {
+                format!(", {}B total", self.total_bytes_removed)
+            } else {
+                let (bytes, unit) = human_readable_bytes(self.total_bytes_removed);
+                format!(", {bytes:.1}{unit} total")
+            }
+        };
+        self.config.shell().status(
+            status,
+            format!(
+                "{} files/directories{byte_count}",
```

Comment:
I included directories since I wanted it to communicate something if it only deleted directories (instead of saying "0 files"), since that wouldn't accurately reflect what it did. I agree the phrasing is a little awkward. It could possibly track them separately (like `N files, M directories, XMB total`). Or it could special case it so that it normally just says `N files, XMB total`, but if there are 0 files and nonzero directories, it could say `0 files, M directories`.

Reply:
That makes sense.

I like the idea of listing out directories separately (or even only showing them if there are no files) but I'll leave it to you for whether that is a future improvement or to do it now.  The incremental improvement of that is not worth blocking this.

Reply:
I made a change so that it tracks files and directories separately. I decided to not display the number of directories since it feels kinda noisy to me, but it will display the directory count if no files were found. I suspect that will almost never happen, though, since cargo always places files in the `target` directory, and I don't suspect cache directories to ever be missing files.